### PR TITLE
Fix core language tests

### PR DESCRIPTION
**Pull request by anupama-pathirage**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: tests/ballerina-unit-test/src/test/resources/test-src/balo/test_balo/types/finite_type_test.bal [hunk 1]
```diff
@@ -88,11 +88,15 @@ function finiteAssignmentRefValueTypeCaseTwo() returns foo:POrInt {
...
 
 function testFiniteTypeWithMatch() returns foo:PreparedResult {
-    match foo() {
-        foo:PreparedResult x => return x;
-        () => return "qqq";
-        error => return "qqq";
+    var x = foo();
+    if (x is foo:PreparedResult) {
+        return x;
+    } else if (x is ()) {
```
Comment (MaryamZi) [inline tests/ballerina-unit-test/src/test/resources/test-src/balo/test_balo/types/finite_type_test.bal]: can every check from this point be in a single `else` block?

File: tests/ballerina-unit-test/src/test/resources/test-src/balo/test_balo/types/finite_type_test.bal [hunk 2]
```diff
@@ -88,11 +88,15 @@ function finiteAssignmentRefValueTypeCaseTwo() returns foo:POrInt {
 }
 
 function testFiniteTypeWithMatch() returns foo:PreparedResult {
-    match foo() {
-        foo:PreparedResult x => return x;
-        () => return "qqq";
-        error => return "qqq";
+    var x = foo();
+    if (x is foo:PreparedResult) {
```
Comment (VijithaEkanayake) [inline tests/ballerina-unit-test/src/test/resources/test-src/balo/test_balo/types/finite_type_test.bal]: Can we simplify this as follows?

if (x is foo:PreparedResult) {
    return x;
} else {
   return "qqq";
}

File: tests/ballerina-unit-test/src/test/resources/test-src/balo/test_balo/types/finite_type_test.bal (hunk 1 above)
Comment (anupama-pathirage) [reply tests/ballerina-unit-test/src/test/resources/test-src/balo/test_balo/types/finite_type_test.bal]: yes  +1

File: tests/ballerina-unit-test/src/test/resources/test-src/balo/test_balo/types/finite_type_test.bal (hunk 2 above)
Comment (anupama-pathirage) [reply tests/ballerina-unit-test/src/test/resources/test-src/balo/test_balo/types/finite_type_test.bal]: Yes + 1

**Pull request merged**